tests/test_dispatch_types.py

```diff
@@ -0,0 +1,160 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Tests for the frequenz.client.dispatch types."""
+
+from datetime import datetime, timedelta, timezone
+
+
+def test_component_selector() -> None:
+    """Test the component selector."""
+    # pylint: disable=import-outside-toplevel
+    from frequenz.client.common.microgrid.components.components import ComponentCategory
+
+    from frequenz.client.dispatch._types import (
+        ComponentSelector,
+        _component_selector_from_protobuf,
+        _component_selector_to_protobuf,
+    )
+
+    assert ComponentSelector is not None
+
+    for selector in (
+        [1, 2, 3],
+        [10, 20, 30],
+        ComponentCategory.BATTERY,
+        ComponentCategory.GRID,
+        ComponentCategory.METER,
+    ):
+        protobuf = _component_selector_to_protobuf(selector)  # type: ignore
+        assert _component_selector_from_protobuf(protobuf) == selector
```

Comment:
Any reason not to use `pytest.mark.parametrize` for this? It will move some clutter out of the test itself.

Maybe you can borrow some ideas from here if you have more complex tests and you want, for example, to assign names to them, etc.:

https://github.com/frequenz-floss/pymdownx-superfence-filter-lines-python/blob/231d34c4f80ad00785b231d49e28453b5ccfb08e/tests/test_do_validate.py#L33-L127

Reply:
Sure, you don't have to make it more complicated, but using `parametrize()` is actually simpler, you don't need to create classes for test cases, that was just a FYI in case it was useful. Just using parametrize is as simple as:

```py
from frequenz.client.common.microgrid.components.components import ComponentCategory

from frequenz.client.dispatch._types import (
    ComponentSelector,
    _component_selector_from_protobuf,
    _component_selector_to_protobuf,
)

@pytest.mark.parametrize("selector", [
        [1, 2, 3],
        [10, 20, 30],
        ComponentCategory.BATTERY,
        ComponentCategory.GRID,
        ComponentCategory.METER,
    ])
def test_component_selector(selector: list[int] | ComponentCategory) -> None:
    """Test the component selector."""
    protobuf = _component_selector_to_protobuf(selector)  # type: ignore
    assert _component_selector_from_protobuf(protobuf) == selector
```

Another advantage of using parametrize is it creates one test per parameter, so you get exactly which parameter failed and you can execute the test for one particular parameter only, also without it, the test will be aborted after the first failure, while using parametrize will continue testing other parameters (unless the fail early option is used).

If you think updating these tests here now are not worth it, I'm OKish with that, but I strongly suggest using `parametrize()` in future tests.